### PR TITLE
Allow pre-generated urls to be passed in rss feeds

### PR DESCRIPTION
**Pull request by zadeviggers**
## Changes

- Allows pre-generated URLs to be passed as the link for rss items.
- Previously, passing an already valid URL as the link would result in an invalid URL being generated in the RSS feed.

## Testing

Added extra section to rss feed test to check if pre-generated urls work.

## Docs

Bug fix only.

### CODE REVIEW

File: packages/astro/src/core/ssr/rss.ts [hunk 1]
```diff
@@ -47,9 +47,12 @@ export function generateRSS(args: GenerateRSSArgs): string {
 		if (typeof result !== 'object') throw new Error(`[${srcFile}] rss.items expected an object. got: "${JSON.stringify(result)}"`);
 		if (!result.title) throw new Error(`[${srcFile}] rss.items required "title" property is missing. got: "${JSON.stringify(result)}"`);
 		if (!result.link) throw new Error(`[${srcFile}] rss.items required "link" property is missing. got: "${JSON.stringify(result)}"`);
+		let itemLink = result.link;
+		// If the item's link is already a valid URL, don't mess with it.
+		if (!/^(?:https?:\/\/)?(www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b(?:[-a-zA-Z0-9()@:%_\+.~#?&//=]*)$/i.test(result.link)) itemLink = canonicalURL(result.link, site).href;
```
Comment (natemoo-re) [inline packages/astro/src/core/ssr/rss.ts:52]: Complicated RegExps like this scare me from a maintenance perspective. 

Could this be simplified to use something like...
```
const isValidURL = (value) => {
  try {
    new URL(result.link)
    return true;
  } catch (e) {}
  return false;
}
```

**Review by natemoo-re — COMMENTED** [review]
Looks good! Would love to support this but the URL RegExp scares me 😅

File: packages/astro/src/core/ssr/rss.ts (hunk 1 above)
Comment (zadeviggers) [reply packages/astro/src/core/ssr/rss.ts:52]: Quite possibly, but I wasn't sure which would be more performant.

**Comment by jonathantneal** [comment]
Hey @zadeviggers, thanks for this. @natemoo-re can help you with tests tomorrow. Stay tuned!

**Comment by natemoo-re** [comment]
Hey @zadeviggers can I get push permissions for your fork? You uncovered a bug with the `rss` function but I have fixed it! Just need permissions to push my fix.

**Comment by natemoo-re** [comment]
I'm going to close this in favor of #2443 which also has these commits, plus some fixes.

**Pull request closed without merging**